lib/matplotlib/tests/test_backend_gtk3.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from matplotlib import pyplot as plt
+
+import pytest
+
+
+pytest.importorskip("matplotlib.backends.backend_gtk3agg")
+
+
+@pytest.mark.backend("gtk3agg")
+def test_correct_key():
+    if os.environ.get("TRAVIS"):
+        pytest.xfail("Test is blocking on Travis.")
+
+    from gi.repository import Gdk, Gtk
+    fig = plt.figure()
+    buf = []
+
+    def send(event):
+        for key, mod in [
+                (Gdk.KEY_a, Gdk.ModifierType.SHIFT_MASK),
+                (Gdk.KEY_a, 0),
+                (Gdk.KEY_a, Gdk.ModifierType.CONTROL_MASK),
+                (Gdk.KEY_agrave, 0),
+                (Gdk.KEY_Control_L, Gdk.ModifierType.MOD1_MASK),
+                (Gdk.KEY_Alt_L, Gdk.ModifierType.CONTROL_MASK),
+                (Gdk.KEY_agrave,
+                 Gdk.ModifierType.CONTROL_MASK
+                 | Gdk.ModifierType.MOD1_MASK
+                 | Gdk.ModifierType.MOD4_MASK),
+                (0xfd16, 0),   # KEY_3270_Play.
+                (Gdk.KEY_BackSpace, 0),
+                (Gdk.KEY_BackSpace, Gdk.ModifierType.CONTROL_MASK),
+        ]:
+            # This is not actually really the right API: it depends on the
+            # actual keymap (e.g. on Azerty, shift+agrave -> 0).
+            Gtk.test_widget_send_key(fig.canvas, key, mod)
```

Comment:
Could this be why it's getting stuck? Maybe you can check `len(buf)` and assert the result after shutting down?

Reply:
Actually turns out that even locally this is not passing anymore, i.e. calls to Gtk.test_widget_send_event (or Gdk.simulate_key_event) don't result in the key_press_event handler being called at all.  At some point it worked, so likely a Gtk version thing?  I just xfailed the whole thing for now...

Reply:
According to https://mail.gnome.org/archives/commits-list/2017-January/msg01173.html, those were never fully implemented (I only see an x11 version in that commit).